auth/session API call and console logging out the response is having an error.

janie-lee-developer opened the pull request:
<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/open-sauced/blob/HEAD/CONTRIBUTING.md#create-a-pull-request.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/open-sauced/blob/HEAD/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

<!-- 
Please do not leave this blank 
This PR [adds/removes/fixes/replaces] the [feature/bug/etc]. 
-->
I am calling an API call auth/session, and trying to console.log the response from the API call, but I am getting this error. It's saying I am trying to render an obj inside a jsx element. I don't think I am doing that. Also, this branch is up to date with the remote hot repo's beta branch. In case the error might be from some other issue.

## Related Tickets & Documents
<!-- 
Please use this format link issue numbers: Fixes #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 
-->
#366 

## Mobile & Desktop Screenshots/Recordings

<!-- Visual changes require screenshots -->


## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?



<!-- note: PRs with deleted sections will be marked invalid -->

bdougie commented:
@janie-lee-developer canyou paste the error message here too?

bdougie commented on src/components/PrimaryNav.tsx:
```diff
@@ -29,13 +29,19 @@ const StarTheRepo = (): JSX.Element => (
 );
 
-const PrimaryNav = (): JSX.Element => {
+const PrimaryNav = async (): Promise<JSX.Element> => {
```
This is the error. async is not needed is not ideal. Instead `useEffect`. Here is an example https://github.com/open-sauced/hot/blob/beta/src/components/RepoWrap.tsx#L48-L50

https://docs.react-async.com/

janie-lee-developer replied:
Hi Brian! Thank you! I removed async on the component, PrimaryNav, and that error is gone now!

I am still having following 2 errors.
1. the API call is returning 401 status.
2.  npm run format fails, saying that I need to catch the error, so I added the try and error keywords, but It's still failing the npm run format... Could you please help?
<img width="1388" alt="Screen Shot 2022-10-12 at 3 52 10 PM" src="https://user-images.githubusercontent.com/58411107/195435605-7fa947d0-ea84-4056-b770-8fa0e5c9c16b.png">
<img width="611" alt="Screen Shot 2022-10-12 at 3 51 10 PM" src="https://user-images.githubusercontent.com/58411107/195435613-2fd3e0ff-ec09-40fc-98b7-7da1b1be0a20.png">

bdougie replied:
Following up from my DM. 

1. Check if you are passing the token or jwt in the call to the API - 401 unauthorized makes me assume not.
2. add a .catch to your function - similar to here: https://github.com/open-sauced/hot/blob/b801823659e89ef81f7eb14682901a20275b8990/src/hooks/useVotedRepos.ts#L18-L20

janie-lee-developer replied:
Hi Brian, thank you so much for the help during the meeting! I am now getting 200 successfully, but for some reason, when I run npm run format, I am still getting an error saying that 'Promise must be awaited'.
<img width="595" alt="Screen Shot 2022-10-13 at 2 08 22 PM" src="https://user-images.githubusercontent.com/58411107/195674032-3adfdb2b-5c7d-484c-b3d5-1a581f638959.png">
<img width="595" alt="Screen Shot 2022-10-13 at 2 06 40 PM" src="https://user-images.githubusercontent.com/58411107/195674054-ef13a7a7-a2b0-4814-8780-a2ab448bdc2e.png">

<img width="1574" alt="Screen Shot 2022-10-13 at 2 06 50 PM" src="https://user-images.githubusercontent.com/58411107/195674060-ef46a4da-3266-4e52-9be3-a85c85e232ba.png">

SunGoldTech replied:
Hey Janie-lie, this error is as a result of execution of an async function without the await keyword... adding the `await` keyword to the `fetchAuthSection()` function call should fix. And on a second thought, the function should return the api's call response/error and not `console.log` response/error.

janie-lee-developer commented:
Hi Brian, could you advise what I can do to fix the error that I am getting from npm run format? I just want to finish up this part before I move on to submissions API.

janie-lee-developer commented:
@bdougie 
So I used the useEffect to useSession, but followings are changes I made.
1. @ useSession needed to use useSupabaseAuth, so I removed this and replaced with /lib/supabase.
2. added context folder at root level to create the global-state.tsx file.
3. created interfaces folder at root level to create the global-state-types.ts file.
4.  I removed the onboarded and waitlisted properties being added to the global context when useSession is ran.

Getting 404 error. 

<img width="975" alt="Screen Shot 2022-10-18 at 4 06 10 PM" src="https://user-images.githubusercontent.com/58411107/196533673-a5b6e9b5-4eb2-4a46-bd04-ecbf2212817d.png">

0-vortex commented:
> 1. @ useSession needed to use useSupabaseAuth, so I removed this and replaced with /lib/supabase.

This call doesn't need to be cached, it's safe to call every time in supabaseAuth

> 2. added context folder at root level to create the global-state.tsx file.
> 3. created interfaces folder at root level to create the global-state-types.ts file.
> 4. I removed the onboarded and waitlisted properties being added to the global context when useSession is ran.

This is not needed nor desired as part of this particular issue, since it is a big scope creep, creating a global state [1] and then caching that session data [2]. They are great things to do, when the users can accurately log in.

You can keep this draft open as it will ideally solve an additional 2 more issues, however the code itself needed for this is just 1 line fetching the `auth/session` api in supabaseAuth hook.

> Getting 404 error.
> 
> <img alt="Screen Shot 2022-10-18 at 4 06 10 PM" width="975" src="https://user-images.githubusercontent.com/58411107/196533673-a5b6e9b5-4eb2-4a46-bd04-ecbf2212817d.png">

That error says the avatar image for the user is missing, it's unrelated.

bdougie commented:
> You can keep this draft open as it will ideally solve an additional 2 more issues, however, the code itself needed for this is just 1 line fetching the auth/session api in supabaseAuth hook.

@janie-lee-developer - open a new PR with the previous commit that just calls auth/session. As @0-vortex mentioned, leave this one open for solving the other things later, but for now. Just calling that endpoint is enough to move on to submissions.

janie-lee-developer commented:
Okay! Understood! Thank you Ted and Brian. Will do!

bdougie commented:
Closing this now. I think we have done what is needed to unblock submissions and handle errors.
